pkg/apis/contrail/v1alpha1/command_types.go

```diff
@@ -81,6 +83,16 @@ func (c *Command) PodsCertSubjects(podList *corev1.PodList, serviceIP string) []
 	return PodsCertSubjects(podList, c.Spec.CommonConfiguration.HostNetwork, altIPs)
 }
 
+//Upgrading is used to check if command is in the upgrade process
+func (c *Command) Upgrading() bool {
+	if c.Status.UpgradeState == CommandShuttingDownBeforeUpgrade ||
+		c.Status.UpgradeState == CommandUpgrading ||
+		c.Status.UpgradeState == CommandUpgradeFailed {
```

Comment:
if upgrade failed then Command is still upgrading? Isn't failure of the upgrade the end of the upgrade?

Reply:
changed